pkgs/applications/editors/pulsar/update.mjs

```diff
@@ -0,0 +1,84 @@
+#!/usr/bin/env nix-shell
+/*
+#!nix-shell -i "node --experimental-fetch" -p nodejs-18_x
+*/
+
+import { promises as fs } from 'node:fs';
+import { promisify } from 'node:util';
+import { exec as _exec } from 'node:child_process';
+const exec = promisify(_exec);
+
+const constants = {
+    githubUrl: "https://api.github.com/repos/pulsar-edit/pulsar/releases",
+    sha256FileURL: (newVersion) => `https://github.com/pulsar-edit/pulsar/releases/download/v${newVersion}/SHA256SUMS.txt`,
+    x86_64FileURL: (newVersion) => `https://github.com/pulsar-edit/pulsar/releases/download/v${newVersion}/Linux.pulsar-${newVersion}.tar.gz`,
+    aarch64FileURL: (newVersion) => `https://github.com/pulsar-edit/pulsar/releases/download/v${newVersion}/ARM.Linux.pulsar-${newVersion}-arm64.tar.gz`,
+};
+
+async function getLatestVersion() {
+    const requestResult = await fetch(constants.githubUrl);
+    if (!requestResult.ok) {
+        console.error("Failed to fetch releases");
+        console.error(requestResult);
+        process.exit(1);
+    };
+    let jsonResult = await requestResult.json();
+
+    jsonResult = jsonResult.filter((release) => !release.prerelease && !release.draft);
+    if (jsonResult.length == 0) {
+        console.error("No releases found");
+        process.exit(1);
+    }
+
+    return jsonResult[0].tag_name.replace(/^v/, '');
+}
+
+async function getSha256Sum(url) {
+    return exec(`nix-prefetch-url --unpack ${url}`). // --unpack is needed to get the hash of the decompressed tarball
+        then(({ stdout, _ }) => stdout.trim()). // stdout is the hash of the decompressed tarball
+        then((stdout) => exec(`nix --experimental-features nix-command hash to-sri --type sha256 "${stdout}"`)). // Convert the hash to SRI format
+        then(({ stdout, _ }) => stdout.trim()) // stdout is the SRI hash
+}
+
+async function getSha256Sums(newVersion) {
+    // Upstream provides a file with the hashes of the files, but it's not in the SRI format, and it refers to the compressed tarball
```

Comment:
note that fetchurl can be used as a source aswell and it will be unpacked as needed

Reply:
Doesn't using `fetchurl` requires a custom `unpackPhase` ?

Reply:
tar.gz and such is already handeled by the existing unpackphase afaik. zip requirest unzip in nativebuildinputs